test/performance/perf-kinesis/src/mz.rs

```diff
@@ -81,7 +81,16 @@ pub async fn query_materialized_view_until(
                 }
             }
         }
-        tokio::time::delay_for(Duration::from_secs(1) - timer.elapsed()).await;
+
+        let elapsed = timer.elapsed().as_millis();
+        if elapsed < 1000 {
+            tokio::time::delay_for(Duration::from_millis((1000 - elapsed) as u64)).await;
+        } else {
+            log::info!(
+                "Expected to query for records in 1000 milliseconds, took {}",
+                elapsed
+            );
+        }
```

Comment:
Good catch! As a general rule, though, moving out of the domain of durations just to move back into it should be avoided when possible! You should be able to just operate on durations directly here?

```
let elapsed = timer.elapsed();
if elapsed <= Duration::from_secs(1) {
    tokio::time::delay_for(Duration::from_secs(1) - elapsed).await;
}  else {
    log::info!("Expected to query for records in 1s, took {:?}", elapsed);
}
```

Reply:
Fixed here!: https://github.com/MaterializeInc/materialize/pull/3186